http/basic/src/main/java/io/cloudevents/http/HttpMessageFactory.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2018-Present The CloudEvents Authors
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.cloudevents.http;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.core.format.EventFormat;
+import io.cloudevents.core.message.MessageReader;
+import io.cloudevents.core.message.MessageWriter;
+import io.cloudevents.core.message.impl.GenericStructuredMessageReader;
+import io.cloudevents.core.message.impl.MessageUtils;
+import io.cloudevents.core.message.impl.UnknownEncodingMessageReader;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+public class HttpMessageFactory {
+
+    public static MessageReader createReader(Map<String,String> headers, byte[] body) throws IllegalArgumentException {
+        return MessageUtils.parseStructuredOrBinaryMessage(
+            () -> headers.get("Content-Type"),
+            format -> new GenericStructuredMessageReader(format, body),
+            () -> headers.get(CloudEventsHeaders.SPEC_VERSION),
+            sv -> new HttpMessageReader(sv, headers.entrySet(), body),
+            UnknownEncodingMessageReader::new
+        );
+    }
+
+    public static MessageWriter createWriter(BiConsumer<String, String> putHeader, Consumer<byte[]> putBody) {
+        return new HttpMessageWriter(putHeader, putBody);
+    }
+
+    public static CloudEvent readCloudEvent(Map<String,String> headers, byte[] body) {
```

Comment:
Do we really need `readCloudEvent`/`writeCloudEvent`? The user goes from/to event with another single method invocation...